main.go

```diff
@@ -97,6 +109,28 @@ func runMain() error {
 	}
 }
 
+func shouldTurnOff(minutesUntilActive int) bool {
+	mutex.Lock()
+	stayOnNow := time.Now().Before(stayOnUntil)
+	mutex.Unlock()
+	if stayOnNow {
+		return false
+	}
+	return minutesUntilActive > 15
+}
```

Comment:
A common pattern is to use defer with mutexes. Then you don't need the intermediate variable and it means the unlock doesn't get forgotten as the code evolves.

For example:

```go
func shouldTurnOff(minutesUntilActive int) bool {
    mutex.Lock()
    defer mutex.Unlock()
    if time.Until(stayOnUntil) > 0 {
       return false
    }
    return minutesUntilActive > 15
}